extensions/support/policy/deployment/src/main/java/org/apache/camel/quarkus/component/support/policy/deployment/PolicyProcessor.java

```diff
@@ -0,0 +1,112 @@
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import org.apache.camel.quarkus.core.deployment.UnbannedReflectiveBuildItem;
+import org.jboss.jandex.AnnotationTarget.Kind;
+import org.jboss.jandex.DotName;
+
+class PolicyProcessor {
+
+    private static final String FEATURE = "camel-policy";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    UnbannedReflectiveBuildItem unbanReflectives() {
+        /**
+         * A list of classes annotated with <code>@UriParams</code> which we accept to be registered for reflection
+         * mostly because there are errors when they are removed. TODO: solve the underlying problems and remove as
+         * many entries as possible from the list.
+         */
+        return new UnbannedReflectiveBuildItem(
+                "org.apache.camel.support.processor.DefaultExchangeFormatter",
+                "org.apache.camel.component.pdf.PdfConfiguration",
+                "org.apache.camel.component.netty.NettyConfiguration",
+                "org.apache.camel.component.netty.NettyServerBootstrapConfiguration",
+                "org.apache.camel.component.fhir.FhirUpdateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirOperationEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirConfiguration",
+                "org.apache.camel.component.fhir.FhirLoadPageEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirSearchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirTransactionEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCreateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirValidateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirReadEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCapabilitiesEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirHistoryEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirMetaEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirPatchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirDeleteEndpointConfiguration",
+
+                /* org.apache.camel.component.consul.* can be removed after the upgrade to Camel 3.1 */
+                "org.apache.camel.component.consul.ConsulConfiguration",
+                "org.apache.camel.component.consul.ConsulClientConfiguration",
+                "org.apache.camel.component.consul.health.ConsulHealthCheckRepositoryConfiguration",
+                "org.apache.camel.component.consul.cloud.ConsulServiceRegistryConfiguration");
```

Comment:
It does not change much from the previous implementation as when we'll remove i.e. `PdfConfiguration` from the pdf extension we have to remember to get to this artefact and change it.

I really don't get why this should not be side-by-side with the code declaring the need for the reflective classes.

Reply:
These are the anomalies we need to tackle. They will hopefully disappear soon. If some of these turns out to be more stable, the item can be moved to the respective extension.

Reply:
I give up :)